### PR TITLE
[YAML] Add JDBC branding transforms

### PR DESCRIPTION
**Pull request by Polber**
This PR adds default Read/Write transforms for various JDBC flavors that abstracts out driver management.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @robertwb

File: sdks/java/io/jdbc/build.gradle [hunk 1]
```diff
@@ -33,6 +33,10 @@ dependencies {
   implementation library.java.joda_time
   implementation "org.apache.commons:commons-pool2:2.11.1"
   implementation library.java.slf4j_api
+  implementation 'org.postgresql:postgresql:42.6.0'
```
Comment (robertwb) [inline sdks/java/io/jdbc/build.gradle:36]: I'm not sure we want to add all of these to the base jdbc package--I think the intent was for the user to provide their own dependencies. 

On the other hand, I can see it'd be really handy for cross-langauge users not to have to specify this. Perhaps these dependencies could be added to the expansion service instead?

File: sdks/python/apache_beam/yaml/yaml_provider.py [hunk 2]
```diff
@@ -700,9 +700,25 @@ def __init__(self, transforms, mappings, underlying_provider, defaults=None):
     for transform in transforms.keys():
       if transform not in mappings:
         raise ValueError(f'Missing transform {transform} in mappings.')
-    self._mappings = mappings
+    self._mappings = self.get_mappings_from_transform(mappings)
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/yaml_provider.py:703]: Maybe call this "validate_mappings" or "expand_mappings" to make the intent more clear?

**Review by robertwb — COMMENTED** [review]
Thanks. Just some minor questions.

File: sdks/java/io/jdbc/build.gradle (hunk 1 above)
Comment (Polber) [reply sdks/java/io/jdbc/build.gradle:36]: Yeah I wasn't sure the best spot to put them, but I agree putting them in the expansion service jar makes more sense.

File: sdks/java/expansion-service/build.gradle [hunk 3]
```diff
@@ -48,6 +48,11 @@ dependencies {
   implementation library.java.vendored_grpc_1_54_0
   implementation library.java.vendored_guava_32_1_2_jre
   implementation library.java.slf4j_api
+  // Stage default drivers for JdbcIO schema transforms
```
Comment (robertwb) [inline sdks/java/expansion-service/build.gradle:51]: This is the base class that everyone using an expansion service depends on (even if they have nothing to do with jdbc or even IOs at all). Let's add the dependency to the specific sdks:java:extensions:schemaio-expansion-service.

**Review by robertwb — APPROVED** [review]
Only one minor change.

File: sdks/java/expansion-service/build.gradle (hunk 3 above)
Comment (Polber) [reply sdks/java/expansion-service/build.gradle]: That makes sense, moved.

**Pull request merged**